src/com.unity.git.api/Api/Git/Tasks/GitCleanTask.cs

```diff
@@ -0,0 +1,38 @@
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Unity.VersionControl.Git.Tasks
+{
+    public class GitCleanTask : ProcessTask<string>
+    {
+        private const string TaskName = "git clean";
+        private readonly string arguments;
+
+        public GitCleanTask(IEnumerable<string> files, CancellationToken token,
+            IOutputProcessor<string> processor = null) : base(token, processor ?? new SimpleOutputProcessor())
+        {
+            Guard.ArgumentNotNull(files, "files");
+            Name = TaskName;
+
+            arguments = "clean ";
+            arguments += " -f ";
+            arguments += " -q ";
+
+            foreach (var file in files)
```

Comment:
Git treats directories specially .. 
we need to test that as the documentation states : 
```If the Git configuration variable clean.requireForce is not set to false, git clean will refuse to delete files or directories unless given -f, -n or -i. Git will refuse to delete directories with .git sub directory or file unless a second -f is given.```

Reply:
Do you think we should enable that behaviour? For reference: I checked vscode and it doesn't do that.

Reply:
no i am just saying we need to test the command "git clean -f -q" against added / renamed directories ..

I tried yesterday on a newly added folder with some random files .. and the command did not delete the directory .. unless i specify "-d" as an argument.

Reply:
I wasn't intending to use this with directories, but I can add that support too.
I'll add tests for added/renamed files.

Reply:
ok, given that we explicitly delete every untracked file then it's ok